yts/test_suite_test.go

```diff
@@ -123,110 +123,135 @@ func runTest(t *testing.T, testPath string) {
 
 		if expectError {
 			if unmarshalErr == nil {
-				t.Errorf("Test: %s\nDescription: %s\nError: Expected unmarshal error but got none", testPath, testDescription)
+				t.Errorf(
+					"Test: %s\nDescription: %s\nError: Expected unmarshal error but got none",
+					testPath, testDescription)
 			}
 			return
 		}
 		if unmarshalErr != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Unexpected unmarshal error: %v", testPath, testDescription, unmarshalErr)
+			t.Errorf(
+				"Test: %s\nDescription: %s\nError: Unexpected unmarshal error: %v",
+				testPath, testDescription, unmarshalErr)
 		}
 	})
 
 	t.Run("EventComparisonTest", func(t *testing.T) {
 		shouldSkipTest(t)
-		expectedEventsPath := filepath.Join(testPath, "test.event")
-		if _, err := os.Stat(expectedEventsPath); err != nil {
-			return
-		}
-		expectedEventsBytes, err := os.ReadFile(expectedEventsPath)
-		if err != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Failed to read test.event: %v", testPath, testDescription, err)
-			return
-		}
-		expectedEvents := normalizeLineEndings(string(expectedEventsBytes))
 		expectedEvents = strings.TrimSuffix(expectedEvents, "\n")
 		actualEvents, eventErr := getEvents(inYAML)
 
 		if expectError {
 			if eventErr == nil {
-				t.Errorf("Test: %s\nDescription: %s\nError: Expected error on event parsing but got none", testPath, testDescription)
+				t.Errorf(
+					"Test: %s\nDescription: %s\nError: Expected error on event parsing but got none",
+					testPath, testDescription)
 			}
 			return
 		}
 		if eventErr != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Unexpected error on event parsing: %v", testPath, testDescription, eventErr)
+			t.Errorf("Test: %s\nDescription: %s\nError: Unexpected error on event parsing: %v",
+				testPath, testDescription, eventErr)
 			return
 		}
 		actualEventsStr := normalizeLineEndings(actualEvents)
 		if actualEventsStr != expectedEvents {
-			t.Errorf("Test: %s\nDescription: %s\nError: Event mismatch\nExpected:\n%q\nGot:\n%q", testPath, testDescription, expectedEvents, actualEventsStr)
+			t.Errorf(
+				"Test: %s\nDescription: %s\nError: Event mismatch\nExpected:\n%q\nGot:\n%q",
+				testPath, testDescription, expectedEvents, actualEventsStr)
 		}
 	})
 
-	// Only proceed with marshal and JSON tests if unmarshal was successful and no expected error
-
-	t.Run("MarshalTest", func(t *testing.T) {
+	t.Run("MarshalCanonicalTest", func(t *testing.T) {
 		shouldSkipTest(t)
-		var currentUnmarshaledValue any
-
-		currentUnmarshalErr := yaml.Unmarshal(inYAML, &currentUnmarshaledValue)
+		var unmarshaledValue interface{}
```

Comment:
please use `any`, it is shorter and more readable

```suggestion
		var unmarshaledValue any
```